planning/planning_debug_tools/include/planning_debug_tools/util.hpp

```diff
@@ -63,6 +63,49 @@ inline std::vector<double> getVelocityArray(const T & points)
   return v_arr;
 }
 
+template <class T>
+inline std::vector<double> getAccelerationArray(const T & points)
+{
+  std::vector<double> segment_wise_a_arr;
+  for (size_t i = 0; i < points.size() - 1; ++i) {
+    const auto & prev_point = points.at(i);
+    const auto & next_point = points.at(i + 1);
+
+    const double delta_s = tier4_autoware_utils::calcDistance2d(prev_point, next_point);
+    if (delta_s == 0.0) {
+      segment_wise_a_arr.push_back(0.0);
+    } else {
+      const double prev_vel = getVelocity(prev_point);
+      const double next_vel = getVelocity(next_point);
+
+      const double acc = (std::pow(next_vel, 2) - std::pow(prev_vel, 2)) / 2.0 / delta_s;
+
+      segment_wise_a_arr.push_back(acc);
+    }
+  }
+
+  std::vector<double> point_wise_a_arr;
+  for (size_t i = 0; i < points.size(); ++i) {
+    if (i == 0) {
+      point_wise_a_arr.push_back(segment_wise_a_arr.at(i));
```

Comment:
@takayuki5168 
I think this if clause condition is bit too many, Is this equal to?
```
    if (i == 0 || i == points.size() - 1 || i == points.size() - 2) {
      point_wise_a_arr.push_back(segment_wise_a_arr.at(0));
    } else{
      point_wise_a_arr.push_back((segment_wise_a_arr.at(i - 1) + segment_wise_a_arr.at(i)) / 2.0);
    }
```